.github/workflows/testing.yml

```
@@ -0,0 +1,85 @@
+ name: Continuous Integration
+ 
+ on:
+   push:
+     branches: [main]
+   pull_request:
+     branches:
+       - "*"
+   schedule:
+     - cron: '59 23 * * *'
+   workflow_dispatch:
+    inputs:
+      version:
+        description: Manual CI Run
+        default: test
+        required: false
+
+ jobs:
+   tests:
+     name: ${{ matrix.os }}, ${{ matrix.environment-file }}
+     runs-on: ${{ matrix.os }}
+     timeout-minutes: 30
+     strategy:
+       matrix:
+         os: [ubuntu-latest]
+         environment-file: [
+            ci/py310_sgeop-oldest.yaml,
+            ci/py311_sgeop-latest.yaml,
+            ci/py312_sgeop-latest.yaml,
+            ci/py312_sgeop-dev.yaml,
+         ]
+         include:
+           - environment-file: ci/py312_sgeop-latest.yaml
+             os: macos-13 # Intel
+           - environment-file: ci/py312_sgeop-latest.yaml
+             os: macos-14 # Apple Silicon
+           - environment-file: ci/py312_sgeop-latest.yaml
+             os: windows-latest
+       fail-fast: false
+
+     defaults:
+       run:
+         shell: bash -l {0}
+
+     steps:
+       - name: checkout repo
+         uses: actions/checkout@v4
+         with:
+           fetch-depth: 0 # Fetch all history for all branches and tags.
+       
+       - name: setup micromamba
+         uses: mamba-org/setup-micromamba@v2
+         with:
+           environment-file: ${{ matrix.environment-file }}
+           micromamba-version: "latest"
+
+       # - must be editable install for current tests to pass
+       # - due to data & pathing; probably need to:
+       #    1. rethink current 'convenience' funcs;
+       #    2. move `data/` into `core/data/`
+       #    3. something else
+       - name: install package
+         run: "pip install -e .[tests]"
```

```suggestion
         run: "pip install -e . --no-deps"
```

we should not install test deps with pip

---

Good point for here. That was needed for the original implementation.